crates/wysiwyg/src/composer_model/format.rs

```diff
@@ -648,4 +666,13 @@ mod test {
         model.strike_through();
         assert_eq!(model.state.dom.to_string(), "<del>a</del>bcd<del>ef</del>");
     }
+
+    #[test]
+    fn format_and_unformat_empty_selection() {
+        let mut model = cm("AAA |");
+        model.bold();
+        assert_eq!(tx(&model), "AAA <strong>{~}|</strong>");
```

Comment:
What if I change selection from this state and move back to it afterwards ? 
I suppose we need to make Rust "cheat" a bit so it selects the `~` when needed ? 
Probably worth an additional test.

Reply:
I tried to get `replace_same_node` to remove zero-width whitespaces at the beginning of a TextNode, but that broke list, so I had to revert it. It's probably worth its own PR.